org.eclipse.transformer/src/main/java/org/eclipse/transformer/action/impl/SignatureRuleImpl.java

```diff
@@ -181,6 +197,37 @@ public BundleData getBundleUpdate(String symbolicName) {
 	public String getDirectString(String initialValue) {
 		return directStrings.get(initialValue);
 	}
+        
+        private final Map<String, Map<String, String>> perClassConstantStrings;
+
+        @Override
+        public String getConstantString(String initialValue, String clazz) {
+            Map<String, String> m = perClassConstantStrings.get(clazz);
+            if (m != null) {
+                // Do we have a direct mapping?
+                String full = m.get(initialValue);
+                if (full != null) {
+                    System.out.println("DIRECT REPLACEMENT:[" + clazz + "]" + initialValue + "=>" + full);
```

Comment:
The presumption is that this is a temporary output, and that the usual logger will eventually be used.

Reply:
Yes, Done.